lib/ex_plasma/payment_v1_builder.ex

```diff
@@ -0,0 +1,106 @@
+defmodule ExPlasma.PaymentV1Builder do
+  @moduledoc """
+  Helper module to make crafting plasma transactions much simpler.
+  """
+
+  alias ExPlasma.Output
+  alias ExPlasma.Transaction
+  alias ExPlasma.Transaction.Type.PaymentV1
+  alias ExPlasma.Transaction.TypeMapper
+
+  @output_type TypeMapper.output_type_for(:output_payment_v1)
+
+  @doc """
+  Create a new Transaction
+
+  ## Example
+
+  # Empty payment v1 transaction
+  iex> new()
+  %ExPlasma.Transaction.Type.PaymentV1{inputs: [], outputs: [], metadata: <<0::256>>}
+
+  # New payment v1 transaction with metadata
+  iex> new(metadata: <<1::256>>)
+  %ExPlasma.Transaction.Type.PaymentV1{inputs: [], outputs: [], metadata: <<1::256>>}
+  """
+  @spec new(list()) :: PaymentV1.t()
+  def new(opts \\ []), do: struct(%PaymentV1{}, opts)
+
+  @doc """
+  Adds an input to the Transaction
+
+  ## Example
+
+  iex> new()
+  ...> |> add_input(blknum: 1, txindex: 0, oindex: 0)
+  ...> |> add_input(blknum: 2, txindex: 0, oindex: 0)
+  %ExPlasma.Transaction.Type.PaymentV1{
+    inputs: [
+      %ExPlasma.Output{output_id: %{blknum: 1, txindex: 0, oindex: 0}},
+      %ExPlasma.Output{output_id: %{blknum: 2, txindex: 0, oindex: 0}},
+    ]
+  }
+  """
+  @spec add_input(PaymentV1.t(), keyword()) :: PaymentV1.t()
+  def add_input(txn, opts) do
+    input = %Output{output_id: Enum.into(opts, %{})}
+    %{txn | inputs: txn.inputs ++ [input]}
+  end
+
+  @doc """
+  Adds an output to the Transaction
+
+  ## Example
+
+  iex> new()
+  ...> |> add_output(output_data: %{output_guard: <<1::160>>, token: <<0::160>>, amount: 1})
+  ...> |> add_output(output_guard: <<1::160>>, token: <<0::160>>, amount: 2)
+  %ExPlasma.Transaction.Type.PaymentV1{
+    outputs: [
+      %ExPlasma.Output{output_type: 1, output_data: %{output_guard: <<1::160>>, token: <<0::160>>, amount: 1}},
+      %ExPlasma.Output{output_type: 1, output_data: %{output_guard: <<1::160>>, token: <<0::160>>, amount: 2}},
+    ]
+  }
+  """
+  @spec add_output(PaymentV1.t(), list()) :: PaymentV1.t()
+  def add_output(txn, output_data: data) do
```

Comment:
if we assume that future output types will always have an `output_data`, we could generalize this and have it take a type as well:

```suggestion
  def add_output(txn, output_type: 1, output_data: data) do
```

Also a suggestion for the future not necessarily for this PR.

Reply:
Yeah I guess we need to see how it evolve in the future, we can make this a bit more flexible and rework the builder as we add new types.